### ISSUE TITLE
Don't use System.out and System.err for logging (in non test classes)

### ISSUE THREAD

**Issue by JMayrbaeurl** [Azure/azure-iot-sdk-java#62]
- **OS and version used:** Windows 10 

- **Java runtime used:** 1.7 

- **SDK version used:** 1.0.21 

# Description of the issue:
In over hundred of cases System.out.println or System.err.println is used in combination with a logger (Custom logger implementation) to log information in the source code. This is not necessary, since the logger can be configured this way. And the default case, logging to the console, leads to having log information written twice.

# Code sample exhibiting the issue:

# Console log of the issue:

**Comment by tameraw** [comment]
@JMayrbaeurl - Thanks for bringing this up. At this point we will not be able to address this right away. But we do want to fix it. If you're able to provide a PR for this, we'll take it.

**Comment by JMayrbaeurl** [comment]
Just to make sure and avoid any kind of misunderstandings with the PR (Sorry but my last PRs were all declined several times and and I want to avoid this the next time): I'll remove any kind of System.out.println and System.err.println when a CustomLogger method call with the same message is available. What should I do when the messages are different?

E.g. in com.microsoft.azure.sdk.iot.device.transport.amqps.AmqpsTransport.protonMessageToIoTHubMessage(MessageImpl)
```java
                catch (IllegalAccessException e)
                {
                    logger.LogError(e);
                    System.err.println("Attempted to access private or protected member of class during message conversion.");
                }
```
Or
```java
                catch (InvocationTargetException e)
                {
                   logger.LogError(e); 
                   System.err.println("Exception thrown while attempting to get member variable. See: " + e.getMessage());
                }

```
BTW. Keeping the System.out.println or System.err.println shouldn't be an option!

**Comment by prmathur-microsoft** [comment]
It would make sense to remove the message from System.err.println and move it to the log and provide details to the log that can be interpreted when read later.

**Comment by prmathur-microsoft** [comment]
Fix checked in @3d69666f8a4502a705b05bfadc29a294396666fa
Please re-open if you see any further issues.

**Comment by jemackayoe** [comment]
I am seeing logging in the console when the client is attempting a reconnect to the IoT hub. The message is "Lost connection to the server. Reconnecting x time." This should be controllable by log level not simply println().
